### PR TITLE
aio: reimplement unit tests

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: aio/src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { async, inject, ComponentFixture, TestBed } from '@angular/core/testing';
+import { APP_BASE_HREF } from '@angular/common';
 import { AppComponent } from './app.component';
 import { AppModule } from './app.module';
+import { SearchService } from 'app/search/search.service';
```
Comment (gkalpak) [inline aio/src/app/app.component.spec.ts]: How come this works? I would expect `./search/search.service`?

File: aio/src/app/app.component.spec.ts [hunk 2]
```diff
@@ -23,4 +26,38 @@ describe('AppComponent', () => {
   it('should create', () => {
     expect(component).toBeDefined();
   });
+
+  describe('isHamburgerVisible', () => {
```
Comment (gkalpak) [inline aio/src/app/app.component.spec.ts]: I don't think empty describe blocks show up as pending when you run the tests, do they?
If I want them to show up, I add an empty `it('should have tests')` block inside.

File: aio/src/app/documents/document.service.spec.ts [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
 
 describe('DocumentService', () => {
+
+  let injector: ReflectiveInjector;
+
   beforeEach(() => {
-    TestBed.configureTestingModule({
-      providers: [DocumentService]
-    });
+    injector = ReflectiveInjector.resolveAndCreate([
```
Comment (gkalpak) [inline aio/src/app/documents/document.service.spec.ts]: What do we gain by not using `TestBed`?

File: aio/src/app/layout/nav-menu/nav-menu.component.spec.ts [hunk 4]
```diff
@@ -8,7 +8,8 @@ describe('NavMenuComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ NavMenuComponent ]
+      declarations: [ NavMenuComponent ],
```
Comment (gkalpak) [inline aio/src/app/layout/nav-menu/nav-menu.component.spec.ts]: Inconsistent whitespace around array elements 😱

File: aio/src/app/navigation/navigation.service.spec.ts [hunk 5]
```diff
@@ -1,14 +1,34 @@
...
 
-  it('should ...', inject([NavigationService], (service: NavigationService) => {
+  it('should be creatable', () => {
+    const service: NavigationService = injector.get(NavigationService);
     expect(service).toBeTruthy();
-  }));
+  });
+
+  xit('should fetch the navigation views', () => {});
+  xit('should compute the navigation map', () => {});
```
Comment (gkalpak) [inline aio/src/app/navigation/navigation.service.spec.ts]: Did you try using `it('...')` (without the second argument)?

File: aio/src/app/shared/location.service.spec.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { Location, LocationStrategy } from '@angular/common';
 import { MockLocationStrategy } from '@angular/common/testing';
 import { LocationService } from './location.service';
 
-fdescribe('LocationService', () => {
```
Comment (gkalpak) [inline aio/src/app/shared/location.service.spec.ts]: 😱 We should have checks that catch this stuff.

**Review by gkalpak — APPROVED** [review]
A couple of nits. Nothing blocking.

File: aio/src/app/app.component.spec.ts (hunk 1 above)
Comment (petebacondarwin) [inline aio/src/app/app.component.spec.ts]: It is a feature of the webpack setup in Angular CLI :-): `app` is an alias.

File: aio/src/app/app.component.spec.ts (hunk 2 above)
Comment (petebacondarwin) [inline aio/src/app/app.component.spec.ts]: That was what I was complaining about at the sync meeting. If I remove the function then the tests crashed.

File: aio/src/app/documents/document.service.spec.ts (hunk 3 above)
Comment (petebacondarwin) [inline aio/src/app/documents/document.service.spec.ts]: I find the tests look simpler, IMO. I don't have to think about modules and all that goes along with that. I think that TestBed is still the key to testing things that a related to the compiler, e.g. components, pipes and directives etc.

File: aio/src/app/layout/nav-menu/nav-menu.component.spec.ts (hunk 4 above)
Comment (petebacondarwin) [inline aio/src/app/layout/nav-menu/nav-menu.component.spec.ts]: 🙀

File: aio/src/app/navigation/navigation.service.spec.ts (hunk 5 above)
Comment (petebacondarwin) [inline aio/src/app/navigation/navigation.service.spec.ts]: yes

File: aio/src/app/shared/location.service.spec.ts (hunk 6 above)
Comment (petebacondarwin) [inline aio/src/app/shared/location.service.spec.ts]: feel free to add it to the Travis build :-)

File: aio/src/app/documents/document.service.spec.ts (hunk 3 above)
Comment (gkalpak) [inline aio/src/app/documents/document.service.spec.ts]: I wouldn't say they look simpler. They look more low-level-y :smiley:
 TBH, both seem equally simple, but `TestBed` feels more idiomatic. I'm fine either way. Especially if `resolveAndCreate()` is faster than creating a whole module.

Comment (petebacondarwin) [inline aio/src/app/documents/document.service.spec.ts]: I feel that TestBed feels more magical for little benefit in the scenario of testing a service

File: aio/src/app/layout/nav-menu/nav-menu.component.spec.ts (hunk 4 above)
Comment (petebacondarwin) [inline aio/src/app/layout/nav-menu/nav-menu.component.spec.ts]: I'll fix this when fixing up the nav-menu in general later today

**Pull request closed without merging**